tests/formatter/test_project.py

```diff
@@ -14,6 +14,42 @@ def test_project_dependencies(fmt: Fmt) -> None:
     fmt(fmt_project, start, expected)
 
 
+def test_project_dependencies_with_double_quotes(fmt: Fmt) -> None:
+    start = """
+    [project]
+    dependencies = [
+        'packaging>=20.0;python_version>"3.4"',
+        "appdirs"
+    ]
+    """
+    expected = """
+    [project]
+    dependencies = [
+      "appdirs",
+      'packaging>=20;python_version>"3.4"',
+    ]
+    """
+    fmt(fmt_project, start, expected)
+
+
+def test_project_dependencies_with_mixed_quotes(fmt: Fmt) -> None:
+    start = """
+    [project]
+    dependencies = [
+        "packaging>=20.0;python_version>\\"3.4\\" and python_version != '3.5'",
+        "appdirs"
+    ]
+    """
+    expected = """
+    [project]
+    dependencies = [
+      "appdirs",
+      "packaging>=20;python_version>\\"3.4\\" and python_version != '3.5'",
```

Comment:
Shouldn't this change it to `'` and remove the quoting part?

Reply:
> would you also consider a PR that uses `packaging.requirements.Requirement` for the normalisation of the PEP 508 requirements? (e.g. guarantee all of them use the same style of quotes)

I'd prefer not to pull in a new dependency, if doable. We can enhance the current logic in place to do this.

Reply:
Currently this change will only opportunistically change the representation from basic TOML strings to literal TOML strings, but it does not change the normalisation per-se.

That is the reason why I asked if you would be happy with using `packaging.requirements.Requirement`. `packaging` normalises all quotes to double quotes. Then  the entire string could be represented as a TOML literal.

(The drawback of using `packaging` for normalisation is that it will introduce a empty space between `;` and the environment markers).

Reply:
> I'd prefer not to pull in a new dependency, if doable. We can enhance the current logic in place to do this.

That is more tricky 😅, I don't know if I am up for this job, since the environment marker syntax is very complex (I am not even sure if it can be represented as a regex).

Reply:
I mean we could just would avoid if possible 🤔 not against it would just want to check first if doable without.

Reply:
Let me try a little experiment to see if we can use some naïve regexes to do it (but I will try in a different PR).

Reply:
I added an experiment in #4